src/EntityFramework6.Npgsql/SqlGenerators/SqlBaseGenerator.cs

```diff
@@ -1191,6 +1209,12 @@ VisitedExpression VisitFunction(EdmFunction function, IList<DbExpression> args,
                         throw new NotSupportedException("cast type name argument must be a constant expression.");
 
                     return new CastExpression(args[0].Accept(this), typeNameExpression.Value.ToString());
+                }else if (functionName == "coalesce")
```

Comment:
How can this code be triggered? Isn't it enough with the code above?

Reply:
You are right! I have made a mistake cleaning code, this fragment is for general case to coalesce translation in other pull request #94